### PR TITLE
Support expanding :find command argument using 'findexpr'

### PR DESCRIPTION
**Pull request by yegappan**
When the 'findexpr' option is set, use that to expand the argument to the ':find' command.

### CODE REVIEW

**Comment by chrisbra** [comment]
thanks!

**Comment by bfrg** [comment]
It seems like `v:fname` is always prefixed with a `^`. For example, if I type `:find foo` and press TAB, `v:fname` will be set to `^foo`.

**Comment by zeertzjq** [comment]
There is also another problem:
- When cmdline is `a*b`, `v:fname` is set to `^a.*b`
- When cmdline is `a.*b`, `v:fname` is set to `^a\..*b`

File: src/ex_docmd.c [hunk 1]
```diff
@@ -6979,6 +6967,61 @@ eval_findexpr(char_u *ptr, int len)
...
+
+    *numMatches = 0;
+    *files = NULL;
+
+    // File name expansion uses wildchars.  But the 'findexpr' expression
+    // expects a regular expression argument.  So convert wildchars in the
+    // argument to regular expression patterns.
+    regpat = file_pat_to_reg_pat(pat, NULL, NULL, FALSE);
+    if (regpat == NULL)
+	return FAIL;
```
Comment (zeertzjq) [inline src/ex_docmd.c:6990]: This isn't correct. The docs for 'findexpr' doesn't require it to expect a regular expression argument.

**Comment by zeertzjq** [comment]
Yet another problem: https://github.com/vim/vim/pull/15934#issuecomment-2434169039

**Comment by vim-ml** [comment]
Hi,

On Wed, Oct 23, 2024 at 1:58 PM bfrg ***@***.***> wrote:

> It seems like v:fname is always prefixed with a ^. For example, if I type :find
> foo and press TAB, v:fname will be set to ^foo.
>
>
>
Yes.  When expanding file names, Vim internally adds wildcards to the
search string.
It will be simpler to use in a Vim regular expression comparison if this is
converted to
a regular expression pattern.  So when the 'findexpr' expression is
evaluated for
command line argument completion, the string is converted to a regular
expression.
I should have updated the document to reflect this.

- Yegappan

**Comment by vim-ml** [comment]
On Wed, Oct 23, 2024 at 7:42 PM zeertzjq ***@***.***> wrote:

> There is also another problem:
>
>    - When cmdline is a*b, v:fname is set to ^a.*b
>    - When cmdline is a.*b, v:fname is set to ^a\..*b
>
>
> Yes.  Before evaluating 'findexpr' for command line argument expansion,
the command
argument is converted from a glob pattern to a regular expression pattern.
That is why
you see this behavior.

Regards,
Yegappan

**Pull request closed without merging**